hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestDataSourceDefaults.scala

```diff
@@ -570,8 +570,8 @@ class TestDataSourceDefaults extends ScalaAssertionSupport {
     val laterOrderingVal: Object = laterRecord.get("favoriteIntNumber")
     val newerPayload = new OverwriteWithLatestAvroPayload(laterRecord, HoodieAvroUtils.convertValueForSpecificDataTypes(fieldSchema, laterOrderingVal, false).asInstanceOf[Comparable[_]])
 
-    // it will provide the record with greatest combine value
-    val preCombinedPayload = basePayload.preCombine(newerPayload)
+    // it always returns the latest payload.
+    val preCombinedPayload = newerPayload.preCombine(basePayload)
```

Comment:
Fix validation instead since we need to validate the preCombine to be called on a payload with a smaller ordering field.